### ISSUE TITLE
Compilation error upon installation

### ISSUE THREAD

**Issue by TedLyngmo** [miloyip/rapidjson#874]
Hi!

I just tried installing but ran into problems.

Fedora release 24
Kernel: 4.8.16-200.fc24.x86_64
Total number of processors: 12
```
% git clone https://github.com/miloyip/rapidjson.git
% cd rapidjson
% git submodule update --init
% mkdir build
% cd build
% cmake ..
-- The CXX compiler identification is GNU 6.3.1
-- Check for working CXX compiler: /usr/lib64/ccache/c++
-- Check for working CXX compiler: /usr/lib64/ccache/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.11")
-- Found GTestSrc: /home/ted/git/rapidjson/thirdparty/gtest/googletest
-- The C compiler identification is GNU 6.3.1
-- Check for working C compiler: /usr/lib64/ccache/cc
-- Check for working C compiler: /usr/lib64/ccache/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found PythonInterp: /usr/bin/python (found version "2.7.13")
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ted/git/rapidjson/build
% make
[...]
I get a lot of warnings when generating docs about stuff not being documented. Example:
/home/ted/git/rapidjson/include/rapidjson/document.h:2462: warning: Member ConstArray (typedef) of class rapidjson::GenericArray is not documented.
[...]
Scanning dependencies of target simplepullreader
[ 45%] Building CXX object example/CMakeFiles/simplepullreader.dir/simplepullreader/simplepullreader.cpp.o
/home/ted/git/rapidjson/example/simplepullreader/simplepullreader.cpp: In constructor ‘MyHandler::MyHandler()’:
/home/ted/git/rapidjson/example/simplepullreader/simplepullreader.cpp:15:8: error: ‘MyHandler::type’ should be initialized in the member initialization list [-Werror=effc++]
 struct MyHandler {
        ^~~~~~~~~
/home/ted/git/rapidjson/example/simplepullreader/simplepullreader.cpp:15:8: error: ‘MyHandler::data’ should be initialized in the member initialization list [-Werror=effc++]
/home/ted/git/rapidjson/example/simplepullreader/simplepullreader.cpp: In function ‘int main()’:
/home/ted/git/rapidjson/example/simplepullreader/simplepullreader.cpp:38:15: note: synthesized method ‘MyHandler::MyHandler()’ first required here
     MyHandler handler;
               ^~~~~~~
cc1plus: all warnings being treated as errors
example/CMakeFiles/simplepullreader.dir/build.make:62: recipe for target 'example/CMakeFiles/simplepullreader.dir/simplepullreader/simplepullreader.cpp.o' failed
make[2]: *** [example/CMakeFiles/simplepullreader.dir/simplepullreader/simplepullreader.cpp.o] Error 1
CMakeFiles/Makefile2:1655: recipe for target 'example/CMakeFiles/simplepullreader.dir/all' failed
make[1]: *** [example/CMakeFiles/simplepullreader.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
```
If you need more info, please let me know!

Br,
Ted

**Comment by stervo4ka** [comment]
I wonder why did the maintainers enable this quite exotic warning just to disable it with pragmas in the source code.

**Comment by pablin87** [comment]
I am having the same issue

**Comment by StilesCrisis** [comment]
This issue was resolved with PR #881 and can be closed.
